compiler/cpp/test/ReaderTest.cpp

```diff
@@ -0,0 +1,234 @@
+//
+// Created by Marc on 27.05.2021.
+//
+
+#include <reader/Reader.h>
+#include <gtest/gtest.h>
+
+// -------------------------------------------------------- EOF --------------------------------------------------------
+
+TEST(ReaderTests, EoF) {
+    // Initialize
+    Reader reader = Reader("Test", 2);
+
+    // Test
+    EXPECT_EQ (reader.getLookahead(),  'T');
+    reader.advance();
+    reader.advance();
+    reader.advance();
+    EXPECT_EQ (reader.getLookahead(),  't');
+    reader.advance();
+    EXPECT_EQ (reader.getLookahead(),  EOF);
+    reader.advance();
+    EXPECT_EQ (reader.getLookahead(),  EOF);
+}
+
+TEST(ReaderTests, EoFInEmptyFile) {
+    // Initialize
+    Reader reader = Reader("", 5);
+
+    // Test
+    EXPECT_EQ (reader.getLookahead(),  EOF);
+    reader.advance();
+    EXPECT_EQ (reader.getLookahead(),  EOF);
+    reader.advance();
+    EXPECT_EQ (reader.getLookahead(),  EOF);
+}
+
+// ----------------------------------------------------- Lookahead -----------------------------------------------------
+
+TEST(ReaderTests, Lookahead) {
+    // Initialize
+    Reader reader = Reader("File input", 2);
+
+    // Test
+    EXPECT_EQ (reader.getLookahead(),  'F');
+}
+
+TEST(ReaderTests, LookaheadAdvance) {
+    // Initialize
+    Reader reader = Reader("File input", 2);
+
+    // Test
+    EXPECT_EQ (reader.getLookahead(),  'F');
+    reader.advance();
+    EXPECT_EQ (reader.getLookahead(),  'i');
+    reader.advance();
+    EXPECT_EQ (reader.getLookahead(),  'l');
+    reader.advance();
+    EXPECT_EQ (reader.getLookahead(),  'e');
+}
+
+// --------------------------------------------------- Max lookahead ---------------------------------------------------
+
+TEST(ReaderTests, MaxLookahead) {
+    // Initialize
+    Reader reader = Reader("File input", 4);
+
+    // Test
+    EXPECT_EQ (reader.getMaxLookahead(),  "File");
+}
+
+TEST(ReaderTests, MexLookaheadAdvance) {
+    // Initialize
+    Reader reader = Reader("File input", 3);
+
+    // Test
+    EXPECT_EQ (reader.getMaxLookahead(),  "Fil");
```

Comment:
the name here will be a merge conflict ;)

Reply:
Took care of that ;)